### PR TITLE
Query Wazuh-DB to fill OS cache

### LINKED ISSUE #20761: Query the Wazuh-DB in case the OS cache doesn't contain data for a specific agent
## Description

The new vulnerability detector holds a cache for the OS data of the agents.
But when this data isn't available, the `SocketDBWrapper` must be used to query `wazuh-db` to retrieve this information.

## Subtasks

- [x] Development. ETA ->
- [x] Testing. ETA->
- [x] Final review. ETA->

## DoD

- [x] The requested data is retrieved successfully
- [x] Add UT
- [x] CI Pass

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|Closes #20761 |

## Description

This PR queries the Wazuh-DB when the OS cache of vulnerability detector doesn't contain data for an agent.
The item is added to the cache also.

## Tests

UT tests were added.

Also, a manual test was carried out. Some temporary debug messages were added to confirm the DB was queried successfully just once

```
2023/12/14 03:48:03 wazuh-modulesd:vulnerability-scanner[86147] databaseFeedManager.hpp:134 at processMessage(): INFO: Processing file: queue/vd_updater/tmp/contents/vd_1.0.0_vd_4.8.0_230577_1700904966.json

2023/12/14 03:48:05 wazuh-modulesd:vulnerability-scanner[86147] osDataCache.hpp:128 at getOsData(): WARNING: Os data retrieved from DB for agentId: 004

2023/12/14 03:48:05 wazuh-modulesd:vulnerability-scanner[86147] packageScanner.hpp:154 at handleRequest(): DEBUG: Scanning package: libc-dev-bin, CNA: canonical

2023/12/14 03:48:06 wazuh-modulesd:vulnerability-scanner[86147] osDataCache.hpp:118 at getOsData(): WARNING: Os data found in cache for agentId: 004
```

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Source installation

- [x] Added unit tests (for new features)

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/osDataCache.hpp [hunk 1]
```diff
@@ -13,10 +13,16 @@
...
 #include "cacheLRU.hpp"
+#include "loggerHelper.h"
 #include "singleton.hpp"
+#include "socketDBWrapper.hpp"
+#include "vulnerabilityScannerDefs.hpp"
+#include "wazuhDBQueryBuilder.hpp"
 #include <shared_mutex>
 #include <string>
 
+auto constexpr OS_CACHE_WDB_SOCKET {"queue/db/wdb"};
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/osDataCache.hpp:24]: Maybe the constant WDB_LOCAL_SOCK from defs.h could be used directly without duplicating the definition.

**Review by matias-braida — COMMENTED** [review]
GJ

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/osDataCache.hpp (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/osDataCache.hpp:24]: I'm afraid it wasn't possible.
The `defs.h` file contains an **USER** definition that conflicts with `rocksDB`

```
In file included from /home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/../scanOrchestrator/osDataCache.hpp:16,
                 from /home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/../scanOrchestrator/scanContext.hpp:19,
                 from /home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:16,
                 from /home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.cpp:12:
/home/vagrant/wazuh/src/headers/defs.h:100:25: error: expected identifier before string constant
  100 | #define USER            "wazuh"
      |                         ^~~~~~~
/home/vagrant/wazuh/src/external/rocksdb/include/rocksdb/thread_status.h:47:5: note: in expansion of macro ‘USER’
   47 |     USER,               // User thread (Non-RocksDB BG thread)
      |     ^~~~
/home/vagrant/wazuh/src/headers/defs.h:100:25: error: expected ‘}’ before string constant
  100 | #define USER            "wazuh"
      |                         ^~~~~~~
/home/vagrant/wazuh/src/external/rocksdb/include/rocksdb/thread_status.h:47:5: note: in expansion of macro ‘USER’
   47 |     USER,               // User thread (Non-RocksDB BG thread)
      |     ^~~~
```

**Comment by pereyra-m** [comment]
@matias-braida , please review again.
I've added some changes to fix the UT and moved the initialization of the socket DB wrapper.

**Review by matias-braida — APPROVED** [review]
LGTM

**Pull request merged**